From d32c89e61881a3d123ea9d51a2e6f36c2aac76e6 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 20 Apr 2012 02:48:24 +0200
Subject: [PATCH] transaction: avoid garbage collecting the anchor job

Make sure the anchor job is never considered garbage, even if it has no links
leading to it (this will be allowed in the next patch).
(cherry picked from commit 38809d9dfed4c75d9e97c4e5da2ff957723c4cad)
---
 src/transaction.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/transaction.c b/src/transaction.c
index cac58e6..ddb02c0 100644
--- a/src/transaction.c
+++ b/src/transaction.c
@@ -454,7 +454,7 @@ static void transaction_collect_garbage(Transaction *tr) {
                 again = false;
 
                 HASHMAP_FOREACH(j, tr->jobs, i) {
-                        if (j->object_list) {
+                        if (tr->anchor_job == j || j->object_list) {
                                 /* log_debug("Keeping job %s/%s because of %s/%s", */
                                 /*           j->unit->id, job_type_to_string(j->type), */
                                 /*           j->object_list->subject ? j->object_list->subject->unit->id : "root", */
